includes/reports/reports-functions.php

```diff
@@ -1053,50 +1192,131 @@ function default_display_charts_group( $report ) {
+	$range_select = EDD()->html->select(
+		array(
+			'name'             => 'range',
+			'class'            => 'edd-graphs-date-options',
+			'options'          => $range_options,
+			'variations'       => false,
+			'show_option_all'  => false,
+			'show_option_none' => false,
+			'selected'         => $selected_range,
+		)
+	);
+
+	$relative_range_select = EDD()->html->select(
+		array(
+			'name'             => 'relative_range',
+			'class'            => 'edd-graphs-relative-date-options',
+			'options'          => $relative_range_options,
+			'variations'       => false,
+			'show_option_all'  => false,
+			'show_option_none' => false,
+			'selected'         => $selected_relative_range,
+		)
+	);
 
 	// From.
-	$from = EDD()->html->date_field( array(
-		'id'          => 'filter_from',
-		'name'        => 'filter_from',
-		'value'       => ( empty( $dates['from'] ) || ( 'other' !== $dates['range'] ) ) ? '' : $dates['from'],
-		'placeholder' => _x( 'From', 'date filter', 'easy-digital-downloads' ),
-	) );
+	$from = EDD()->html->date_field(
+		array(
+			'id'          => 'filter_from',
+			'name'        => 'filter_from',
+			'value'       => ( empty( $dates['from'] ) || ( 'other' !== $dates['range'] ) ) ? '' : $dates['from'],
+			'placeholder' => _x( 'From', 'date filter', 'easy-digital-downloads' ),
+		)
+	);
 
 	// To.
-	$to = EDD()->html->date_field( array(
-		'id'          => 'filter_to',
-		'name'        => 'filter_to',
-		'value'       => ( empty( $dates['to'] ) || ( 'other' !== $dates['range'] ) ) ? '' : $dates['to'],
-		'placeholder' => _x( 'To', 'date filter', 'easy-digital-downloads' ),
-	) );
+	$to = EDD()->html->date_field(
+		array(
+			'id'          => 'filter_to',
+			'name'        => 'filter_to',
+			'value'       => ( empty( $dates['to'] ) || ( 'other' !== $dates['range'] ) ) ? '' : $dates['to'],
+			'placeholder' => _x( 'To', 'date filter', 'easy-digital-downloads' ),
+		)
+	);
 
 	// Output fields
-	?><span class="edd-date-range-picker graph-option-section"><?php
-		echo $range;
-	?></span>
+	?>
+	<div class="edd-date-range-picker graph-option-section" data-range="<?php echo esc_attr( $selected_range ); ?>">
+		<?php echo $range_select; ?>
+		<!-- DATE RANGES -->
+		<div class="edd-date-range-dates">
+			<span class="dashicons dashicons-calendar edd-date-main-icon"></span>
+			<div class="edd-date-range-selected-date">
+				<?php
+				foreach ( $range_options as $range_key => $range_name ) :
+					$range_dates          = \EDD\Reports\parse_dates_for_range( $range_key );
+					$selected_range_class = ( $selected_range !== $range_key ) ? 'hidden' : '';
+					?>
+					<span class="<?php echo esc_attr( $selected_range_class ); ?>" data-range="<?php echo esc_attr( $range_key ); ?>"><?php echo esc_html( edd_get_edd_timezone_equivalent_date_from_utc( $range_dates['start'] )->format( $date_format ) ); ?> - <?php echo esc_html( edd_get_edd_timezone_equivalent_date_from_utc( $range_dates['end'] )->format( $date_format ) ); ?></span>
```

Comment:
This is a minor quibble, but if the start and end dates are identical (because I picked today or yesterday or a single date), could this show just the one date instead of the `from - to`? I feel like that would be nice:

![image](https://user-images.githubusercontent.com/4853066/190678045-d37a1442-08df-4db3-bab6-2b3cfc23338c.png)

Also, and I don't know if this is possible, but when I change this date selector to something that's a different range, it would be nice if the "compared to" dropdown reset to the previous period instead of staying whatever it was--I switched from a quarter range to today and the comparison was still showing "previous quarter".